server/services/v1/query_runner.go

```diff
@@ -580,11 +583,15 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 		}
 
 		resp.Facets = rowReader.getFacets()
-		ps := int64(pageSize)
-		totalPages := int32(rowReader.getTotalFound()/ps) + 1
+		if totalPages == nil {
+			ps := float64(pageSize)
+			tp := int32(math.Ceil(float64(rowReader.getTotalFound()) / ps))
```

Comment:
just combine ps with this line `tp := int32(math.Ceil(float64(rowReader.getTotalFound()) / float64(pageSize)))`